File: common/src/main/java/me/fullidle/ficore/ficore/common/api/pokemon/NatureWrapper.java
package me.fullidle.ficore.ficore.common.api.pokemon;

import me.fullidle.ficore.ficore.common.api.Wrapper;

public abstract class NatureWrapper<T> extends Wrapper<T> {
    public NatureWrapper(T original) {
        super(original);
    }

    public abstract String getName();
}

File: v1_21/Cob/src/main/kotlin/me/fullidle/ficore/ficore/vCob_21/NatureWrapper.kt
package me.fullidle.ficore.ficore.vCob_21

import com.cobblemon.mod.common.pokemon.Nature
import me.fullidle.ficore.ficore.common.api.pokemon.NatureWrapper

class NatureWrapper(original: Nature) : NatureWrapper<Nature>(original) {
    override fun getName(): String {
        return original.displayName
    }

    override fun getType(): Class<Nature> {
        return Nature::class.java
    }
}
